src/util/records.py

```diff
@@ -1469,23 +1475,28 @@ def replace_record_references_batch(cr, id_mapping, model_src, model_dst=None, r
         if ir.company_dependent_comodel:
             if ir.company_dependent_comodel == model_src:
                 assert model_src == model_dst
+                json_path = cr.mogrify(
+                    "$.* ? ({})".format(" || ".join(["@ == %s"] * len(id_mapping))),
+                    list(id_mapping),
+                ).decode()
+
                 query = cr.mogrify(
                     format_query(
                         cr,
                         """
                         UPDATE {table}
                            SET {column} = (
-                                   SELECT jsonb_object_agg(key, COALESCE((%s::jsonb->>value)::int, value::int))
+                                   SELECT jsonb_object_agg(key, COALESCE(((jsonb_object(%s::text[]))->>value)::int, value::int))
```

Comment:
```suggestion
                                   SELECT jsonb_object_agg(key, COALESCE(((jsonb_object(%s::text[]))->>value), value)::int)
```

Given that `NULL` can also be cast to `int`, we can do the casting directly on the output of `COALESCE`.